### PR TITLE
FriendshipsFeedRequest

### PR DESCRIPTION
**Pull request by noirtier-villefort**
Added FriendshipsFeedRequest, FriendshipsFeedResponse

### CODE REVIEW

File: lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import io.github.taz03.jia.requests.InstagramGetRequest;
+import io.github.taz03.jia.responses.users.FriendshipsFeedResponse;
+import io.github.taz03.jia.responses.users.UserResponse;
+
+public final class FriendshipsFeedRequest extends InstagramGetRequest<FriendshipsFeedResponse> {
+
+    public FriendshipsFeedRequest(long pk, FriendshipsFeeds action) {
+        super(FriendshipsFeedResponse.class, String.format("/api/v1/friendships/%s/%s/", pk, action.name().toLowerCase()));
+    }
+    public static enum FriendshipsFeeds{
```
Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java:12]: add an empty line here

File: lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java
Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java:1]: add docs

Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java:1]: this should be actually be in its own new package `io.github.taz03.jia.requests.friendships`

File: lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java
Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java:1]: same package naming for response as well

File: lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package io.github.taz03.jia.responses.users;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.github.taz03.jia.responses.InstagramResponse;
+import io.github.taz03.jia.responses.models.user.User;
+
+import java.util.List;
```
Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java:8]: follow the import order  from wiki

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package io.github.taz03.jia.requests.users;
+
+import io.github.taz03.jia.requests.InstagramGetRequest;
+import io.github.taz03.jia.responses.users.FriendshipsFeedResponse;
+import io.github.taz03.jia.responses.users.UserResponse;
+
+public final class FriendshipsFeedRequest extends InstagramGetRequest<FriendshipsFeedResponse> {
+
```
[the remark this answers is not in the document]
Comment (noirtier-villefort) [reply lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java]: removed

File: lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java (hunk 1 above)
Comment (noirtier-villefort) [reply lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java]: added

File: lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java
Comment (noirtier-villefort) [reply lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java:1]: added

Comment (noirtier-villefort) [reply lib/src/main/java/io/github/taz03/jia/requests/users/FriendshipsFeedRequest.java:1]: moved

File: lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java
Comment (noirtier-villefort) [reply lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java:1]: moved to friendships

File: lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java (hunk 2 above)
Comment (noirtier-villefort) [reply lib/src/main/java/io/github/taz03/jia/responses/users/FriendshipsFeedResponse.java]: refactored

File: README.md
[the remark this answers is not in the document]
Comment (noirtier-villefort) [reply README.md:1]: added

File: lib/src/main/java/io/github/taz03/jia/requests/friendships/FriendshipsFeedRequest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package io.github.taz03.jia.requests.friendships;
+
+import io.github.taz03.jia.requests.InstagramGetRequest;
+import io.github.taz03.jia.responses.friendships.FriendshipsFeedResponse;
+
+/**
+ * Represents an Instagram user friendships feed, used to get user followers or followings
```
Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/requests/friendships/FriendshipsFeedRequest.java:7]:
```suggestion
 * Represents an Instagram friendships feed request, used to get user's followers or followings
```

File: lib/src/main/java/io/github/taz03/jia/requests/friendships/FriendshipsFeedRequest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ */
+public final class FriendshipsFeedRequest extends InstagramGetRequest<FriendshipsFeedResponse> {
+    /**
+     * Creates an Instagram friendships feed request
+     * @param pk pk of the user
+     * @param action action of {@linkplain FriendshipsFeeds} to get followers or followings
+     */
```
Comment (Taz03) [inline lib/src/main/java/io/github/taz03/jia/requests/friendships/FriendshipsFeedRequest.java:14]:
```suggestion
    /**
     * Creates an Instagram friendships feed request
     
     * @param pk pk of the user to get friendships of
     * @param action action of {@link FriendshipsFeeds} to get followers or followings
     */
```

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package io.github.taz03.jia.friendships;
+
+import io.github.taz03.jia.requests.friendships.FriendshipsFeedRequest;
+import io.github.taz03.jia.responses.friendships.FriendshipsFeedResponse;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.ExecutionException;
+
+import static io.github.taz03.jia.TestProxy.client;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (Taz03) [inline lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java:10]: import order

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java (hunk 6 above)
Comment (noirtier-villefort) [reply lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java]: fixed

**Review by Taz03 — DISMISSED** [review]

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.concurrent.ExecutionException;
+import io.github.taz03.jia.requests.friendships.FriendshipsFeedRequest;
+import io.github.taz03.jia.responses.friendships.FriendshipsFeedResponse;
+import org.junit.jupiter.api.Test;
+
+public class FriendshipsTest {
+
+    @Test
+    public void friendshipsFeedFollowingTest() throws ExecutionException, InterruptedException {
```
Comment (Taz03) [inline lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java:14]: we don't care about the exact exception so just do `throws Exception`

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        FriendshipsFeedRequest friendshipsFeedFollowing = new FriendshipsFeedRequest(client.getPk(), FriendshipsFeedRequest.FriendshipsFeeds.FOLLOWING);
+        FriendshipsFeedResponse friendshipsFeedFollowingResponse = client.sendRequest(friendshipsFeedFollowing).get();
+
+        assertEquals("ok", friendshipsFeedFollowingResponse.getStatus());
+    }
+
+    @Test
+    public void friendshipsFeedFollowersTest() throws ExecutionException, InterruptedException {
```
Comment (Taz03) [inline lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java:22]: same here

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java
Comment (Taz03) [inline lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java:1]: this file should be named `FriendshipsFeedTest.java`

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java (hunk 7 above)
Comment (noirtier-villefort) [reply lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java]: fixed

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java (hunk 8 above)
Comment (noirtier-villefort) [reply lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java]: fixed

File: lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java
Comment (noirtier-villefort) [reply lib/src/integrationTest/java/io/github/taz03/jia/friendships/FriendshipsTest.java:1]: renamed

**Review by Taz03 — APPROVED** [review]

**Pull request merged**